crates/forge/tests/integration/mock_call.rs

```diff
@@ -0,0 +1,776 @@
+            trait IMockCheckerLibCall<TContractState> {
+                fn get_constant_thing_with_lib_call(ref self: TContractState, class_hash: ClassHash) -> felt252;
+            }
+
+            #[test]
+            fn start_mock_call_with_library_call_has_no_effect() {
+                let mock_checker_class_hash = declare('MockChecker');
+
+                let class_hash = declare('MockCheckerLibCall');
+                let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @ArrayTrait::new() };
+                let contract_address = deploy(prepared).unwrap();
+
+                let mock_ret_data = 421;
+                start_mock_call(contract_address, 'get_thing', mock_ret_data);
+
+                let dispatcher = IMockCheckerLibCallDispatcher { contract_address };
+                let thing = dispatcher.get_constant_thing_with_lib_call(mock_checker_class_hash);
+                assert(thing == 13, thing);
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "MockChecker".to_string(),
+            Path::new("tests/data/contracts/mock_checker.cairo"),
+        )
+        .unwrap(),
+        Contract::from_code_path(
+            "MockCheckerLibCall".to_string(),
+            Path::new("tests/data/contracts/mock_checker_library_call.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run(
+        &test.path().unwrap(),
+        &String::from("src"),
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        &corelib_path(),
+        &test.contracts(&corelib_path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
+
+    assert_passed!(result);
+}
+
+#[test]
+fn start_mock_call_when_contract_not_deployed() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use result::ResultTrait;
+        use option::OptionTrait;
+        use traits::TryInto;
+        use starknet::ContractAddress;
+        use starknet::Felt252TryIntoContractAddress;
+        use snforge_std::{ declare, PreparedContract, deploy, start_mock_call };
+
+        #[starknet::interface]
+        trait IMockChecker<TContractState> {
+            fn get_thing(ref self: TContractState) -> felt252;
+        }
+
+        #[test]
+        fn start_mock_call_when_contract_not_deployed() {
+            let calldata = array![420];
+
+            let class_hash = declare('MockChecker');
+            let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @calldata };
+
+            let contract_address: felt252 = 123;
+            let contract_address: ContractAddress = contract_address.try_into().unwrap();
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+
+            let mock_ret_data = 421;
+            start_mock_call(contract_address, 'get_thing', mock_ret_data);
```

Comment:
I think this test is not finished

Reply:
Dispatcher will not allow to perform a call, it will throw an error

Reply:
Can we use [`library_call_syscall`](https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/system-calls-cairo1/#library_call) then? You can obtain `function_selector` using starknet.py https://starknetpy.readthedocs.io/en/latest/api/hash.html#starknet_py.hash.selector.get_selector_from_name

Reply:
True, we can try to deploy the contract after that and check if it works, should be fine

Reply:
added an issue https://github.com/foundry-rs/starknet-foundry/issues/465